SaraAlert/SaraAlert
# SARAALERT-1693: Make privacy policy link configurable

# Description
Jira Ticket: [SARAALERT-1693](https://tracker.codev.mitre.org/browse/SARAALERT-1693)

There are references to Sara Alert's privacy policy in the messages sent to monitorees. Revise the messages so that the link to a privacy policy is configurable.

This must be done for all languages and jurisdiction custom messages.

## Important Changes
Please list important files (meaning substantial or integral to the PR) along with a list of the general changes that should be highlighted for reviewers.

`.env-prod-enrollment-example`, `config/sara/sara.yml`
- Add support for new `PRIVACY_POLICY_URL` env variable

`app/mailers/patient_mailer.rb`
- If custom message is present, privacy policy link message should only be appended to custom message if link is not present and the privacy policy url is defined
- If custom message is not present, the privacy policy link part of the message if url is defined

`app/views/patient_mailer/assessment_email.html.erb`, `app/views/patient_mailer/closed_email.html.erb`, `app/views/patient_mailer/enrollment_email.html.erb`
- Only include privacy policy section if url is defined

`config/locales`
- Move privacy policy text from footer and sms intro to shared section

## Testing Recommendations

Use rails console with privacy policy env variable set
```
PRIVACY_POLICY_URL=example.org rails c
```

Verify that `ADMIN_OPTIONS['privacy_policy_url']` is set properly
```
2.7.3 :001 > ADMIN_OPTIONS['privacy_policy_url']
 => "example.org" 
```
If not, kill all ruby processes on your machine and retry

1. Testing email
```
Patient.first.update(email: 'patient@example.com', last_assessment_reminder_sent: nil)
PatientMailer.enrollment_email(Patient.first).deliver_now
PatientMailer.assessment_email(Patient.first).deliver_now
PatientMailer.closed_email(Patient.first).deliver_now
```
2. Testing SMS
- Add `puts contents` after `contents` is set (on line 50) of `patient_mailer.rb` and restart rails console
```
Patient.first.update(primary_telephone: '+13333333333')
PatientMailer.enrollment_sms_text_based(Patient.first)
```
Ignore twilio error message and observe printed message
3. Testing custom jurisdiction messages
- Modify privacy policy link in `jurisdiction_messages.yml`
- Update jurisdiction messages with `rake admin:import_or_update_jurisdiction_messages`
- In rails console:
```
PatientMailer.enrollment_sms_text_based(Patient.where(jurisdiction_id: 2).first)
```
- Make sure privacy policy link in custom message is only present once
- Remove privacy policy section of jurisdiction messages with `rake admin:import_or_update_jurisdiction_messages`
- Update jurisdiction messages with `rake admin:import_or_update_jurisdiction_messages`
- In rails console:
```
PatientMailer.enrollment_sms_text_based(Patient.where(jurisdiction_id: 2).first)
```
- Make sure default privacy policy link is appended to custom jurisdiction message
# Checklists

**Submitter:**


@tstrass :


@sgober :


@hackrm :
